x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java

```diff
@@ -241,6 +241,12 @@ private void loadFromSingleLeaf(Block[] blocks, int shard, int segment, BlockLoa
                 }
             }
 
+            SourceLoader sourceLoader = null;
+            if (storedFieldsSpec.requiresSource()) {
+                sourceLoader = shardContexts.get(shard).newSourceLoader.get();
+                storedFieldsSpec = storedFieldsSpec.merge(new StoredFieldsSpec(true, true, sourceLoader.requiredStoredFields()));
```

Comment:
The second parameter, `requiresMetadata`, should be false.

Reply:
Maybe it should be sourced from `storedFieldsSpec` variable (which is always initialized and gets updates by block loaders)? It could be that a block loader required metadata?

Reply:
This is what I meant: e2c77e11120e284c374490846671423507c91b16

Reply:
Yes, that works too. However, I prefer passing `false` for both the first and second arguments, as it yields the same spec since they are XORed in the `merge` method. I think it's clearer to indicate that we are appending `required stored fields` from the source loader to the spec. But I'm okay with your approach as well - just a minor nit.